api/client.go

```diff
@@ -130,8 +131,10 @@ func request[Req, Res any](client Client, method string, path string, query Quer
 	}
 
 	var resBody Res
-	if err := json.Unmarshal(body, &resBody); err != nil {
-		return nil, fmt.Errorf("parsing json response: %w. partial text: %q", err, partialText(body, 100))
+	if reflect.TypeOf(resBody).Size() > 0 {
```

Comment:
This is a fix for the problem where the response body is empty, right? Seems like a good thing to fix here. There's another related problem where the response body might not be JSON if it came from some proxy in-between the client and the API (or the API failed in some unexpected way that did not return JSON), but no reason to fix that other issue as part of this change.

We should be able to avoid reflection here. I believe the error happens when `body` is empty (it doesn't matter so much what type `Res` is). Since `body is a `[]byte` I think this should work:

```suggestion
	if len(body) > 0 {
```

Reply:
What should be behaviour be in that case? If the server did not error, wouldn't an empty success response be appropriate?

Reply:
In that case the client expects an non-empty response and will run into the same json marshalling error. It would be a bug in the server if it erroneously returns an empty response. I guess one could argue it's out of scope of the client to check